cylc/flow/rsync.py

```diff
@@ -0,0 +1,94 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Rsync related functionality."""
+
+from pathlib import Path
+from typing import List, Tuple
+
+import cylc.flow.suite_files as suite_files
+
+
+def get_rsync_rund_cmd(src, dst, reinstall=False, dry_run=False):
+    """Create and return the rsync command used for cylc install/re-install.
+
+    Args:
+        src (str):
+            file path location of source directory
+        dst (str):
+            file path location of destination directory
+        reinstall (bool):
+            indicate reinstall (--delete option added)
+        dry-run (bool):
+            indicate dry-run, rsync will not take place but report output if a
+            real run were to be executed
+
+    Return:
+        list: command to use for rsync.
+
+    """
+
+    rsync_cmd = ["rsync", "-a", "--out-format=%o %n"]
+    if dry_run:
+        rsync_cmd.append("--dry-run")
+    if reinstall:
+        rsync_cmd.append('--delete')
```

Comment:
Consider also using "--no-t --checksum"?
This means that files only get reinstalled if their contents have changed.
Otherwise, they get re-installed if their modification time changes.

Reply:
I don't think the `-no-t` effects the `--checksum` functionality does it?

```
t, --times                 preserve modification times
```